src/core/StelCore.hpp

```diff
@@ -305,6 +305,9 @@ class StelCore : public QObject
 
 	//! Return the observer heliocentric ecliptic position (GZ: presumably J2000)
 	Vec3d getObserverHeliocentricEclipticPos() const;
+	//! Return the observer heliocentric ecliptic velocity. This includes orbital and diurnal motion;
```

Comment:
If you include diurnal changes in velocity, this should be added to the aberration correction of the observer in SolarSystem::computePositions().

Reply:
I suppose then `observerPlanet->getHeliocentricEclipticPos()` should also be replaced with `core->getObserverHeliocentricEclipticPos()`, right? Because currently that computation takes "observer" to be in the center of the planet.

Reply:
Yes, I also think that's all that has to be changed. When I did aberration correction, I said we won't include the diurnal aberration, but when you are doing this computation now, it's even one more step. Not sure about the magnitude, though, it may introduce an observable milliarcsecond wobble.  Of course, switching off topographic correction will also disable that.

Reply:
Do you think it should go right into this PR? This seems quite unrelated.

Reply:
It will be an obvious next change then. Probably we can even make a switch aberration: none/annual/diurnal (for didactic purposes).

Reply:
Why do you only say that it should be applied in `SolarSystem`? Shouldn't it equally affect everything?